test/parallel/test-http2-compat-pushrequest.js

```diff
@@ -0,0 +1,96 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const h2 = require('http2');
+
+function isEven(number) {
+  return number % 2 === 0;
+}
+
+function isOdd(number) {
+  return number % 2 === 1;
+}
+
+// Push a request & response
+
+const server = h2.createServer();
+server.listen(0, common.mustCall(function() {
+  server.once('request', common.mustCall(function(request, response) {
+    assert.ok(isOdd(response.stream.id));
```

Comment:
Since this is only used once, perhaps best to just inline the `response.stream.id % 2 === 1` check